File: 코딩 기초 트레이닝/문자열 잘라서 정렬하기-p181866/문자열 잘라서 정렬하기.js
// https://school.programmers.co.kr/learn/courses/30/lessons/181866

function solution(myString) {
    var answer = [];

    let current = '';
    myString = myString.split("");
    myString.forEach((d, index) => {
        if(d === 'x') {
            if(current.length !== 0) {
                answer.push(current);
                current = '';
            }
        } else {
            current += d;
        }
    });

    if(current.length > 0) answer.push(current);
    answer.sort();

    return answer;
}

File: 코딩 기초 트레이닝/문자열 잘라서 정렬하기-p181866/문자열 잘라서 정렬하기.py
# https://school.programmers.co.kr/learn/courses/30/lessons/181866

def solution(myString):
    answer = []

    current = ''


    for d in list(myString):
        if d == 'x':
            if len(current) != 0:
                answer.append(current)
                current = ''
        else:
            current += d


    if len(current) > 0:
        answer.append(current)

    answer.sort()

    return answer
